hooks/storage/badger/badger.go

```diff
@@ -204,6 +204,12 @@ func (h *Hook) OnSubscribed(cl *mqtt.Client, pk packets.Packet, reasonCodes []by
 			Filter: pk.Filters[i].Filter,
 			Qos:    reasonCodes[i],
 		}
+		if pk.ProtocolVersion == 5 {
```

Comment:
Sorry for the delay! I don't think we need a protocol version check here as the defaults for these values will be the same as the unset values in the struct, if they weren't set (i.e. they were protocol <5).